core/sql/optimizer/ValueDesc.cpp

```diff
@@ -529,14 +529,13 @@ NABoolean ValueId::isAddedColumnWithNonNullDefault() const{
   default:
       break;
   }
-  if (nac && nac->isAddedColumn() && nac->getDefaultValue())
+  if (nac &&  nac->getDefaultValue() && strcmp(nac->getDefaultValue(),"NULL") != 0)
```

Comment:
why is it better? I know what I do work. I don't know about the alternative. But I know how much a re run of regression cost :-)